hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/scm/ReconIncrementalContainerReportHandler.java

```diff
@@ -99,5 +103,6 @@ public void onMessage(final IncrementalContainerReportFromDatanode report,
       }
     }
     containerManager.notifyContainerReportProcessing(false, success);
+    containerSizeCountTask.process(containerManager.getContainers());
```

Comment:
As we discussed, ICR handler get called for every DN, so we need to first check if our task map already contains the container, if yes, then it means , our task will not be impacted just for state as our task maintains count based on size. If container not present in our. map, then only we should call our task. Pls check how complex and feasible this to implement. When ICR handler gets called for deletion of containers, we should verify first after whether containerManager is having the container or not, if not then our task can be called to handle for deletion of container.

Please first write down all use case scenarios related to ICR why and when it is needed. Same for Stale and Dead Node handler.

Reply:
Since If the container does not exist, it creates a new container and adds it to the ContainerManager hence we will have to trigger the ContainerSizeCountTask.

Reply:
I have written in detail about how each of the three handlers work, in this google doc :- 

https://docs.google.com/document/d/1Wz-zeSoSPfdaVyw43GwAyvFNgt0djo5w44j8OjrxJ1I/edit#

Reply:
here the same: this means that the ContainerSizeCountTask is possible to run more times in each defined interval? if we are calling the `process()` here too.

Reply:
Yes @dombizita it will run more number of times then !!